.github/workflows/new-contributor.yml

```diff
@@ -0,0 +1,44 @@
+name: New Contributor Action
+on: 
+    pull_request_target:
+        types:
+          - closed
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+    cancel-in-progress: true
+
+jobs:
+  ping-new-contributor:
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const creator = context.payload.sender.login
+            const opts = github.rest.issues.listForRepo.endpoint.merge({
+              ...context.issue,
+              creator,
+              state: 'all'
+            })
+            const issues = await github.paginate(opts)
+
+            for (const issue of issues) {
+              if (issue.number === context.issue.number) {
+                continue
+              }
+
+              if (issue.pull_request) {
+                return // Creator is already a contributor.
+              }
+            }
```

Comment:
This part lgtm but I am thinking of an alternative

Instead of checking all prs and see if there is any previous prs by the pr creator, I am thinking if we could utilise Github's built-in function to get the current list of contributors and check if the person is inside

https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-contributors

Not sure if this approach would speed up the process since we do not have to go through every closed pr

Reply:
Hi @LamJiuFong, thanks for the suggestion, I have looked into it and I think that it is based on just commits, rather than merged PRs when it considers the list of contributors. I will try to look for other ways to improve performance if possible.

Reply:
Thanks for the changes @KevinEyo1 ! LGTM

Reply:
Hi @LamJiuFong after testing, I realised that since we are running the job on merge, the author is already part of the list by the time the action is run, so it fails to detect the author as a new contributor. I will revert to the original implementation, and I will try to figure out if there can be performance enhancements.

Reply:
Oh I see, sorry for missing that out

What about checking if the author has only one pr merged (which is the newly merged one), would that be possible?

Reply:
@LamJiuFong I am trying to think of a more efficient solution as well but I think to check if the author has only one PR merged we would have to fetch all the PRs anyway (which is what is done in the current implementation) leading to a similar time taken

Reply:
Spoke to Kevin about this previously and he did this check: [Test run to check time](https://github.com/KevinEyo1/markbind/actions/runs/8565425538/job/23473461471?pr=38) which wasn't very long. So I don't think there is a need to overoptimize in this case.